### ISSUE TITLE
Optional step() after console command?

### ISSUE THREAD

**Issue by MSGhero** [HaxeFlixel/flixel#1884]
https://github.com/HaxeFlixel/flixel/blob/dev/flixel/system/debug/console/Console.hx#L247-L251

```haxe
// Step forward one frame to see the results of the command
#if (flash && FLX_DEBUG)
if (FlxG.vcr.paused)
	FlxG.game.debugger.vcr.onStep();
#end
```

I have to say that I really hate this line of code when debugging. Most of the time, I want my game paused on a single frame where I input a series of commands. Sometimes, I just want to see the current value of 2 things, but advancing a frame changes the second value. Sometimes, I use it as a calculator when deciding what to do during that frame. And then sometimes, I write several lines of actual haxe code that I want to run.

If the current behavior should be kept, the only thing I can think of is setting a debugger/console boolean or adding a special character before the string that gets parsed out. Thoughts?

**Comment by Gama11** [comment]
I think it would lead to a lot of confusion if this wasn't the default, because you'd see no visual changes when executing certain commands like `sprite.x = 500`.

Which of the two options you presented do you prefer?

**Comment by thegoodideaco** [comment]
you know, i love that you added the arrow commands including alt and shift.. if those were to use step i would love it even more

**Comment by Gama11** [comment]
@thegoodideaco I'm not sure what you mean.

**Comment by thegoodideaco** [comment]
if I add an item to the watch window, you can step the value by 1, .1 or 10 I believe. But the results are not instant. You have to hit enter to apply it. It would help be able to nudge things perfectly by stepping every time it is changed.

**Comment by thegoodideaco** [comment]
im just going off of what i'm used to from other editors

**Comment by Gama11** [comment]
Ah, that makes sense. But then @MSGhero will want another flag to disable that! :D

**Comment by MSGhero** [comment]
Lol maybe. I was thinking about \ and the only thing that would mess with is regular expressions. No idea if those even work with hscript or if it matters.

**Comment by MSGhero** [comment]
`\` closes the console, so I can't use that. MATLAB uses `...`

**Comment by Gama11** [comment]
For what? Escaping?

Problem is, `...` is valid Haxe syntax in `IntIterator` literals.

**Comment by MSGhero** [comment]
For skipping `onStep`. The text doesn't get parsed, I `substr` it out beforehand. I just need something unique at the end of an input string.
```haxe
var skipStep = text.endsWith("...");
			
if (skipStep)
	text = text.substr( -3);
```
or so.

**Comment by Gama11** [comment]
No, I meant what Matlab uses it for. :)

**Comment by MSGhero** [comment]
Multiline commands, so that what you type doesn't execute immediately.
```matlab
x = sin(theta) * longExpressionThatTakesUpTheWholeScreen + ...
      cos(theta) * stuff;
```

**Comment by Gama11** [comment]
I mean, `...` for this seems okay to me. How often do you anticipate using this? Might become annoying to type?

**Comment by MSGhero** [comment]
Every time I use the console 😄

The other option is a public boolean, like `ConsoleFrontEnd`'s `autoPause`

**Comment by Gama11** [comment]
That seems like a better option to me then if you want this as more of a permanent setting.

**Comment by MSGhero** [comment]
```haxe
// Step forward one frame to see the results of the command
#if (flash && FLX_DEBUG)
if (FlxG.vcr.paused && FlxG.console.stepAfterCommand)
	FlxG.game.debugger.vcr.onStep();
#end
```
Why is it `#if flash`? I don't immediately see anything platform specific.

`ConsoleCommands` has a `#if FLX_DEBUG` wrapping the entire class but also `#if FLX_DEBUG` wrapping internal methods.

**Comment by Gama11** [comment]
Dunno, couldn't get it to work on other targets when I implemented it.

**Comment by MSGhero** [comment]
Looks fine to me without it `¯\_(ツ)_/¯`

**Comment by MSGhero** [comment]
I thought this was supposed to auto close? Maybe the "closes #1884" needs to be in the message body of the PR?

**Comment by Gama11** [comment]
Yeah, it does need to be in the commit message. I didn't realize it was only in the PR's title.
